File: src/main/java/pl/training/workshop/m7/s09_doublenegative/step3/Voucher.java
package pl.training.workshop.m7.s09_doublenegative.step3;

import java.time.LocalDate;

/** Krok 3: isExpired() jest teraz jedyną definicją reguły; isNotExpired() usunięte (Safe Delete). */
public record Voucher(String code, LocalDate validUntil) {
    public boolean isExpired(LocalDate today) {
        return today.isAfter(validUntil);
    }
}

File: csharp/src/Training.Workshop/M7/S09DoubleNegative/Step3/Voucher.cs
namespace Training.Workshop.M7.S09DoubleNegative.Step3;

/// <summary>Krok 3: IsExpired() jest teraz jedyną definicją reguły; IsNotExpired() usunięte (Safe Delete).</summary>
public sealed record Voucher(string Code, DateOnly ValidUntil)
{
    public bool IsExpired(DateOnly today)
    {
        return today > ValidUntil;
    }
}
